### PR TITLE
Converted tests to use cgnsdiff

### PR DESCRIPTION
**Pull request by sseraj**
## Purpose
This is a second attempt at revamping the pyHyp tests, this time using `cgnsdiff` (based on discussion in #52). I generated the ref files by running the examples with 2 procs on the `u20-gcc-ompi-latest` image.

I verified that the tests fail if any of the following change:
- node coordinates (e.g. using different smoothing parameters)
- boundary conditions
- block dimensions

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
Testing steps are on the installation page in the docs.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Comment by sseraj** [comment]
Adding the `Barrier` call fixed the I/O issue with the pyHypMulti tests. The tacc image failure is a separate issue that seems to be memory related. The two tests that fail are the two large 3D cases.

**Comment by sseraj** [comment]
The tacc image error occurs in the `writeCGNS` call, but only when running in parallel. Any ideas on why this might be happening @eirikurj @joanibal?

This did not occur before because all the examples were run in serial. I could use `N_PROCS = 1` but that doesn't solve the underlying problem.

**Comment by sseraj** [comment]
Adding a `barrier` call seems to have fixed a synchronization issue in `writeCGNS` that was only occurring on the TACC image. This PR should be good to go now.

**Review by nwu63 — DISMISSED** [review]
Looks good to me, thanks for putting in the work -- tests look much nicer now. I would like to hear @eirikurj or others' opinions on these extra Barrier calls before merging though.

**Comment by marcomangano** [comment]
@eirikurj is this looking good?

**Comment by sseraj** [comment]
This is still awaiting review @eirikurj @SichengHe

File: src/3D/writeCGNS.F90 [hunk 1]
```diff
@@ -92,6 +92,9 @@ subroutine writeCGNS(fileName)
         ! Copy values and write:
         do k=1,N
 
+           ! Synchronize before scattering
+           call mpi_barrier(hyp_comm_world, ierr)
```
Comment (anilyil) [inline src/3D/writeCGNS.F90:96]: What is the reasoning behind this barrier?

**Comment by anilyil** [comment]
The tests look great, thanks for the work. I am fine with the final barrier before returning on the python level but I am curious about why we have that barrier before the scatter stuff. If that also makes sense, this is good to go for me.

File: src/3D/writeCGNS.F90 (hunk 1 above)
Comment (sseraj) [reply src/3D/writeCGNS.F90:96]: The tests were failing on the TACC image without this barrier. I think it is related to the newer Intel MPI version used on that image. We did not see this problem before because all the tests ran on 1 proc.

Does it make sense to call barrier before a PETSc generalized scatter? I'm not convinced this is a bug in pyHyp as opposed to something wrong with the TACC image or Intel MPI.

Comment (anilyil) [reply src/3D/writeCGNS.F90:96]: Would have expected it to be fine even w/o the barrier but I dont know enough. If it fixes an issue on TACC, then sure.

**Review by anilyil — DISMISSED** [review]

**Comment by anilyil** [comment]
Can you also review again @nwu63 so we can merge this?

File: src/3D/writeCGNS.F90 (hunk 1 above)
Comment (nwu63) [reply src/3D/writeCGNS.F90:96]: Yeah this is a bit weird, Sabet and I had some conversations about this but I don't really understand what's going on. Maybe it's an MPI implementation-specific behaviour?

Comment (anilyil) [reply src/3D/writeCGNS.F90:96]: So I can try to read and understand the scatter stuff and its usage, but it will take a few more days probably. To me this is not too performance critical since these meshes are run once and not in the loop etc.

File: pyhyp/MExt.py [hunk 2]
```diff
@@ -5,14 +5,14 @@
 import sys
 
 
-def _tmp_pkg(directory):
+def _tmp_pkg(dirName):
```
Comment (nwu63) [inline pyhyp/MExt.py:8]: Why did you change this? `directory` isn't a built-in keyword right? I'm fine with renaming just wanted to understand

File: tests/test_all_examples.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+baseDir = os.path.dirname(os.path.abspath(__file__))
+refDir = os.path.join(baseDir, "ref")
+
+
+class TestExamples(unittest.TestCase):
+
+    N_PROCS = 2
+
+    def setUp(self):
+        # Insert the repo root in path so that testflo can import examples
```
Comment (nwu63) [inline tests/test_all_examples.py:15]: FYI this was also a trick I used in order to import examples

**Review by nwu63 — COMMENTED** [review]
Just one comment: we don't actually need `examples/__init__.py` right? AFAIK since py3.3+ we no longer need to have an init file to import, and the `os.path` modifications should be enough.

File: src/3D/writeCGNS.F90 (hunk 1 above)
Comment (nwu63) [reply src/3D/writeCGNS.F90:96]: Yeah I thought scatter is blocking already, but according to @sseraj he tried many combinations and this was the only one that worked consistently.

File: pyhyp/MExt.py (hunk 2 above)
Comment (sseraj) [reply pyhyp/MExt.py]: I changed `dir` to `dirName` at the same time you changed `dir` to `directory`. I guess I won in the merge. I changed it back in any case.

**Review by anilyil — DISMISSED** [review]

File: pyhyp/MExt.py (hunk 2 above)
Comment (nwu63) [reply pyhyp/MExt.py]: I like that better, I'll just remove the last commit and then approve.

**Review by nwu63 — APPROVED** [review]

**Review by anilyil — APPROVED** [review]
